### PR TITLE
Returns kubernetes error in case we can't get secret

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/4628

In case we can't fetch the secret (e.g: due to the RBAC issue I'm fixing next :D ), gsctl output is now:
```
$ gsctl show cluster cx60u
An internal error occurred.
Please try again in a few minutes. If that does not success, please inform the Giant Swarm support team.
```
and the api response is:
```
$ curl -H "Authorization: Bearer $TOKEN" https://api.g8s.gauss.eu-central-1.aws.gigantic.io/v4/organizations/byoc-test/credentials/f7dk39/
{"code":"INTERNAL_ERROR","message":"secrets \"credential-f7dk39\" is forbidden: User \"system:serviceaccount:giantswarm:credentiald\" cannot get secrets in the namespace \"giantswarm\""}
```

the gsctl output previously was:
```
$ gsctl show cluster cx60u
Credential not found
Credentials with the given ID could not be found.
```
which is very misleading

### CODE REVIEW

File: service/searcher/service.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func (c *Service) Search(request Request) (*Response, error) {
 	if err != nil {
 		kubernetesSearchErrorTotal.Inc()
 		c.logger.Log("level", "error", "message", "could not get secret", "stack", fmt.Sprintf("%#v", err))
```
Comment (xh3b4sd) [inline service/searcher/service.go]:
```suggestion
```

Comment (kopiczko) [reply service/searcher/service.go]: ^ That would be also my suggestion.

Comment (xh3b4sd) [reply service/searcher/service.go]: I think Joseph needs more **REQUEST CHANGES**.

Comment (JosephSalisbury) [reply service/searcher/service.go]: lol, joseph needs to not just insta-merge

shout at me if you want me to remove this log line :D

Comment (xh3b4sd) [reply service/searcher/service.go]: I already mentioned that and wanted to make your life as easy as possible. Please remove the logging there. It does not make sense, even though you may did not add it yourself.

**Pull request merged**